### DISCUSSION TITLE
Plenti's content driven component architecture is so cool!

### DISCUSSION THREAD

**Discussion by jamestagal** [plentico/plenti#297] [General]
Hi all,

I saw an interesting Svelte Society Youtube video  on ```Dynamic component loading```- [This Week in Svelte (2023 October 20) - Kit 1.26.0, Svelte 4.2.2, dynamically-loaded components](https://www.youtube.com/watch?v=O13bGtOV-aA&t=1782s) where they discuss and show an implementation using Sveltekit. I immediately thought how elegant and powerful Plenti's implementation of this is because it offers a unique take; using its own style of content driven component architecture.
Particularly when combined with the [```allLayouts```](https://plenti.co/docs/alllayouts) 'magic' prop which allows for the freedom to add/remove/rearrange components and also have complete SSR builds so you still get HTML fallbacks is awesome! 

Thanks @jimafisk and the Plenti team! I appreciate your great work 😄

**Comment by jimafisk** [comment]
Thanks @jamestagal! Content driven components took a bit of work because Svelte doesn't support SSR'ing dynamic components out of the box, see:
- https://github.com/sveltejs/svelte/issues/5501
- https://github.com/sveltejs/svelte/issues/3853
- https://github.com/sveltejs/svelte/issues/958

We added the `allLayouts` magic prop to hack around this limitation and during the build we use global component signatures to rig it all up. Thanks for the kinds words!
